velox/vector/BaseVector.cpp

```diff
@@ -669,36 +669,21 @@ void BaseVector::copy(
     const BaseVector* source,
     const SelectivityVector& rows,
     const vector_size_t* toSourceRow) {
-  // Check if there are rows that do not exist in 'source'. Remove these from
-  // 'ranges'.
-  // TODO Update the callers and remove this logic.
-
+  if (!rows.hasSelections()) {
+    return;
+  }
   std::vector<CopyRange> ranges;
   if (toSourceRow == nullptr) {
-    if (source->size() < rows.end()) {
-      SelectivityVector trimmedRows = rows;
-      trimmedRows.setValidRange(source->size(), rows.end(), false);
-      trimmedRows.updateBounds();
-
-      ranges = toCopyRanges(trimmedRows);
-    } else {
+    VELOX_CHECK_GE(source->size(), rows.end());
       ranges = toCopyRanges(rows);
-    }
   } else {
     ranges.reserve(rows.end());
     rows.applyToSelected([&](vector_size_t row) {
       const auto sourceRow = toSourceRow[row];
-      if (sourceRow >= source->size()) {
-        return;
-      }
+      VELOX_CHECK_GT(source->size(), sourceRow);
```

Comment:
`VELOX_DCHECK_GT`?

Reply:
@Yuhta I added DCHECKS only for the paths that are executed per row, for all other they are executed once per vector so should not affect perf.
Please let me know if you still prefer I change it to DCHECK

Reply:
This particular line is per row I think?

Reply:
aha, good catch! didnt realize it from the tiny code snippet here. Thanks for reviewing!